src/coverage.ts

```diff
@@ -49,13 +66,18 @@ export const coverage = async ({ token, ghToken, summary }: Props) => {
   const comparison = res?.check
   const utoken = res?.metadata?.token
 
+  let patchCoverage: number | undefined
+  const addedLines = await getDiffLines(octokit, owner, repo, compareSha, sha)
+  const addedLineNumbers = addedLines.map(line => parseInt(line.match(/\d+/)?.[0] || '0', 10))
+  const coveredAddedLines = addedLineNumbers.filter(line => summary.lines.found > line) // Simplified coverage check
+  patchCoverage = coveredAddedLines.length / addedLineNumbers.length
+
   const rate = summary.lines.hit / summary.lines.found
-  const diff = comparison
-    ? rate - comparison.linesHit / comparison.linesFound
-    : undefined
+  const diff = comparison ? rate - (comparison.linesHit / comparison.linesFound) : undefined
 
   core.debug(`rate: ${rate}`)
   core.debug(`diff: ${diff}`)
+  core.debug(`patchCoverage: ${patchCoverage}`)
 
   const message = ((): {
```

Comment:
We need to separate this in functions.
Make a `coverageStatus` function that creates the main total coverage status message.
Make a  `patchStatus` function that create the patch coverage status message.

Reply:
done